src/names/namespaces.md

```diff
@@ -110,9 +110,8 @@ r[names.namespaces.sub-namespaces]
 ## Sub-namespaces
 
 r[names.namespaces.sub-namespaces.intro]
-The macro namespace is split into two sub-namespaces: one for [bang-style macros] and one for [attributes].
-When an attribute is resolved, any bang-style macros in scope will be ignored.
-And conversely resolving a bang-style macro will ignore attribute macros in scope.
+The macro namespace is split into three sub-namespaces: [bang-style macros], [attributes], and [derives].
```

Comment:
Two sub-namespaces - 1) bang and 2) everything else.
These two sub-namespaces were created out of necessity, to avoid breakage during macro modularizations, so we didn't want to multiply unnecessary entities.